### PR TITLE
[graphql/rpc] Protocol config impl

### PR DESCRIPTION
**Pull request by oxade**
## Description 

Prot cfg impl with rpc 1.0
Stubs for Gas and Display types

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-graphql-rpc/src/types/gas.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+pub(crate) struct GasInput {
+    pub gas_sponsor: Option<Address>,
+    pub gas_payment: Option<Vec<Object>>,
+    pub gas_price: Option<BigInt>,
+    pub gas_budget: Option<BigInt>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, SimpleObject)]
+pub(crate) struct GasCostSummary {
+    pub computation_cost: Option<BigInt>,
```
Comment (gegaowp) [inline crates/sui-graphql-rpc/src/types/gas.rs:19]: why "Option<>"here? GasCostSummary in sui-types does not have Option wrapper.

File: crates/sui-graphql-rpc/src/types/display.rs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_graphql::*;
+
+#[derive(Clone, Debug, PartialEq, Eq, SimpleObject)]
+pub(crate) struct DisplayEntry {
```
Comment (gegaowp) [inline crates/sui-graphql-rpc/src/types/display.rs:7]: what is this for?

File: crates/sui-graphql-rpc/schema/draft_schema.graphql [hunk 3]
```diff
@@ -421,15 +421,22 @@ type Epoch {
...
-  configs: [ProtocolConfig]
-  config(key: String!): ProtocolConfig
+  featureFlags: [ProtocolConfigFeatureFlag]
+  configs: [ProtocolConfigAttr]
+  config(key: String!): ProtocolConfigAttr
+  featureFlag(key: String!): ProtocolConfigFeatureFlag
 }
 
-type ProtocolConfig {
+type ProtocolConfigAttr {
```
Comment (gegaowp) [inline crates/sui-graphql-rpc/schema/draft_schema.graphql:430]: curious what are the application usecases of reading ProtocolConfigAttr and ProtocolConfigFeatureFlag?

**Review by gegaowp — APPROVED** [review]
looks good overall.

File: crates/sui-graphql-rpc/schema/draft_schema.graphql (hunk 3 above)
Comment (amnn) [reply crates/sui-graphql-rpc/schema/draft_schema.graphql:430]: Feature flags are boolean only, and they are stored in their own structure within `ProtocolConfig` which means that feature flag names can shadow other attribute names.

File: crates/sui-graphql-rpc/src/types/display.rs (hunk 2 above)
Comment (amnn) [reply crates/sui-graphql-rpc/src/types/display.rs:7]: Unrelated to ProtocolConfig -- this is for Object Display Standard support.

File: crates/sui-graphql-rpc/src/types/gas.rs (hunk 1 above)
Comment (amnn) [reply crates/sui-graphql-rpc/src/types/gas.rs:19]: It's for more flexibility within a given RPC version, to make changes -- e.g. it means that the server can start producing a `None` response for this if, for some reason, we no longer want to support it, or we want to rename it, etc.

**Review by amnn — APPROVED** [review]
LGTM!

**Pull request merged**